dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -719,6 +727,176 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
+    public void deleteWorkflow(@ApiParam(hidden = true) @Auth User authUser,
+                                   @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @QueryParam("gitRegistry") TokenType gitRegistry,
+                                   @ApiParam(value = "Git repository path", required = true) @QueryParam("repository") String repository) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        // Get all of the users source control tokens
+        List<Token> scTokens = workflowResource.checkOnBitbucketToken(foundUser)
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        // Delete workflow for a given repository
+        if (scTokens.size() > 0) {
+            final Token gitToken = scTokens.get(0);
+            final String tokenSource = gitToken.getTokenSource().toString();
+
+            String gitUrl = "git@" + tokenSource + ":" + repository + ".git";
+            LOG.info("Deleting " + gitUrl);
+
+            // Create a workflow stub object if necessary
+            final Optional<BioWorkflow> existingWorkflow = workflowDAO.findByPath(tokenSource + "/" + repository, false, BioWorkflow.class);
+            if (existingWorkflow.isEmpty()) {
+                String msg = "No workflow with path " + tokenSource + "/" + repository + " exists.";
+                LOG.error(msg);
+                throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+            } else {
+                BioWorkflow workflow = existingWorkflow.get();
+                checkUser(foundUser, workflow);
+                if (Objects.equals(workflow.getMode(), WorkflowMode.STUB)) {
+                    elasticManager.handleIndexUpdate(existingWorkflow.get(), ElasticMode.DELETE);
+                    workflowDAO.delete(workflow);
+                } else {
+                    String msg = "The workflow with path " + tokenSource + "/" + repository + " cannot be deleted.";
+                    LOG.error(msg);
+                    throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+                }
+            }
+        }
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/registries")
+    @Operation(operationId = "getUserRegistries")
+    @ApiOperation(nickname = "getUserRegistries", value = "Get all of the git registries accessible to the logged in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = TokenType.class, responseContainer = "List")
+    public List<TokenType> getUserRegistries(@ApiParam(hidden = true) @Auth User authUser) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        List<TokenType> scTokens = tokenDAO.findByUserId(foundUser.getId())
```

Comment:
Alternately, you could use `findFirst`. Comment applies to subsequent methods as well.

Reply:
In this case though I want to find multiple registries.